### PR TITLE
OCPBUGS-100301: fix(hostedcluster): handle Unknown status in ClusterVersionFailing inversion

### PR DESCRIPTION
**Pull request by reedcort**
## What this PR does / why we need it:

The ClusterVersionFailing → ClusterVersionSucceeding condition inversion switch (hostedcluster_controller.go:755-762) only handled True and False. When ClusterVersionFailing had Status: Unknown (common during control plane disruptions like cluster-size-override), the switch fell through and set ClusterVersionSucceeding.Status to "" (zero value). The API server permanently rejects this on Status().Update, blocking the entire reconcile loop before Phase 7 (CoreHCPChain). This prevented HC-to-HCP annotation sync, causing request-serving pods to schedule with stale node affinity and remain Pending indefinitely.

This fix adds a default case that maps Unknown (and any other unexpected status) to ConditionUnknown. Applied to both reconcile and reconcileLegacy paths.

## Which issue(s) this PR fixes:

Fixes [OCPBUGS-100301](https://redhat.atlassian.net/browse/OCPBUGS-100301)

## Special notes for your reviewer:

Two production incidents: 105 min and 44 hours downtime ([ROSAENG-62684](https://redhat.atlassian.net/browse/ROSAENG-62684)).

Root cause confirmed by operator logs showing `status.conditions[4].status: Unsupported value: ""` and HCP showing `ClusterVersionFailing: Unknown`.

**Bug reproduced on staging cluster** by stripping ClusterVersion conditions from the guest cluster:
```
oc patch clusterversion version --type=json -p '[{"op":"remove","path":"/status/conditions"}]' --subresource=status
```
This caused the HCCO to set `ClusterVersionFailing: Unknown` on the HCP, which immediately triggered the same error loop observed in production:
```
{"level":"error","msg":"Reconciler error","controller":"hostedcluster","error":"[failed to update status: HostedCluster is invalid: [status.conditions[4].status: Unsupported value: \"\": supported values: \"True\", \"False\", \"Unknown\"..."}
```

## Checklist:
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs.
- [x] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -758,6 +758,8 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 					status = metav1.ConditionFalse
 				case metav1.ConditionFalse:
 					status = metav1.ConditionTrue
+				default:
+					status = metav1.ConditionUnknown
```
Comment (bryan-cox) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go]: **Suggestion:** The identical inversion switch exists in `reconcile_legacy.go:441`. Consider extracting a shared helper to eliminate the duplication:
```go
func invertConditionStatus(s metav1.ConditionStatus) metav1.ConditionStatus {
	switch s {
	case metav1.ConditionTrue:
		return metav1.ConditionFalse
	case metav1.ConditionFalse:
		return metav1.ConditionTrue
	default:
		return metav1.ConditionUnknown
	}
}
```
The project already does this with `computeClusterVersionStatus()` which is shared between both paths. This would also let the test exercise the actual production function instead of re-implementing the logic inline.

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go [hunk 2]
```diff
@@ -3995,6 +3995,96 @@ func TestIsProgressing(t *testing.T) {
...
+		{
+			name:           "When ClusterVersionFailing is Unknown it should produce ClusterVersionSucceeding Unknown",
+			hcpStatus:      metav1.ConditionUnknown,
+			expectedStatus: metav1.ConditionUnknown,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
```
Comment (bryan-cox) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go:4031]: **Suggestion:** The subtests should call `t.Parallel()` — other subtests in this file do (e.g., `TestComputeAWSDefaultSGDeletedCondition`). Since these cases are independent:
```go
t.Run(tc.name, func(t *testing.T) {
	t.Parallel()
	g := NewWithT(t)
```

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go [hunk 3]
```diff
@@ -3995,6 +3995,96 @@ func TestIsProgressing(t *testing.T) {
...
+						case metav1.ConditionTrue:
+							status = metav1.ConditionFalse
+						case metav1.ConditionFalse:
+							status = metav1.ConditionTrue
+						default:
+							status = metav1.ConditionUnknown
+						}
+						hcCVOCondition.Status = status
+					}
+				}
```
Comment (bryan-cox) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go]: **Suggestion:** This test re-implements the inversion switch inline rather than calling `reconcile()` or `reconcileLegacy()`. If the production code is changed but this test isn't updated (or vice versa), the test will still pass while the production path is broken.

If extracting `invertConditionStatus()` is too much for a bug-fix PR, at minimum the test name should clarify it's testing the *logic* in isolation, not the *code path*.

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go [hunk 4]
```diff
@@ -3995,6 +3995,96 @@ func TestIsProgressing(t *testing.T) {
...
+		{
+			name:           "When ClusterVersionFailing is False it should invert to ClusterVersionSucceeding True",
+			hcpStatus:      metav1.ConditionFalse,
+			expectedStatus: metav1.ConditionTrue,
+		},
+		{
+			name:           "When ClusterVersionFailing is Unknown it should produce ClusterVersionSucceeding Unknown",
+			hcpStatus:      metav1.ConditionUnknown,
+			expectedStatus: metav1.ConditionUnknown,
+		},
```
Comment (bryan-cox) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go:4025]: **Suggestion:** Consider adding a fourth test case for `hcpStatus: ""` (empty string). That's the exact zero-value that triggered the original production bug, and it would also hit the `default` branch:
```go
{
	name:           "When ClusterVersionFailing has empty status it should produce ClusterVersionSucceeding Unknown",
	hcpStatus:      metav1.ConditionStatus(""),
	expectedStatus: metav1.ConditionUnknown,
},
```

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go [hunk 5]
```diff
@@ -3995,6 +3995,96 @@ func TestIsProgressing(t *testing.T) {
 	}
 }
 
+func TestCVOConditionInversion(t *testing.T) {
```
Comment (bryan-cox) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go]: Per TESTING.md, test function names should map 1:1 to the function under test. Since this doesn't call a production function directly, the name `TestCVOConditionInversion` is fine as-is. But if you extract the helper, rename to `TestInvertConditionStatus` to match.

**Review by bryan-cox — APPROVED** [review]
The fix is correct and addresses two production incidents.

**Review by joshbranham — COMMENTED** [review]
/lgtm

**Review by bryan-cox — APPROVED** [review]
